packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Models/CSharpMethod.cs

```diff
@@ -29,5 +30,18 @@ public CSharpMethod(MethodSignatureBase signature, MethodBodyStatement body, CSh
             Body = body;
             Kind = kind;
         }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CSharpMethod"/> class with a value expression and method signature.
+        /// </summary>
+        /// <param name="signature">The method signature.</param>
+        /// <param name="expression">The value expression.</param>
+        /// <param name="kind">The method kind <see cref="CSharpMethodKinds"/>.</param>
+        public CSharpMethod(MethodSignatureBase signature, ValueExpression expression, CSharpMethodKinds kind)
```

Comment:
in the implementation of enums, a lot of methods are using an expression as their body, such as
```csharp
public override string ToString() => _value;
```
therefore we added it here.